### ISSUE TITLE
CropOrPad in 2D images adds extra channel

### ISSUE THREAD

**Issue by Linardos** [fepegar/torchio#434]
**🐛Bug**

I wanted to crop some images 

**To reproduce**

```python
import torchio as tio
subject = tio.Subject(
    chest_ct=tio.ScalarImage(data_path / patient_csv['image_file_path'][1]),
    heart_mask=tio.LabelMap(data_path / patient_csv['ROI_mask_file_path'][1]),
)
subject.chest_ct.shape
transform = tio.CropOrPad(
    (256, 256, 1),
    mask_name='heart_mask',
)
transformed = transform(subject)
transformed.chest_ct.shape
```

```python-traceback
(1, 256, 256, 2)
```

**Expected behavior**

I expected a 2D image of shape (1,256,256,1) as CropOrPad only accepts 3 dimensions however when I use (256,256,1) as input the result is an image of shape (1,256,256,2). Screenshot attached for your reference. The extra dimension is basically a blank image.

**Actual behavior**

![image](https://user-images.githubusercontent.com/26694607/106166619-06f9fd00-6195-11eb-86bf-420b15470afe.png)

**Comment by fepegar** [comment]
Hi, @Linardos. Thanks for reporting.

I'm not able to reproduce using this code:

```python
import torch
import torchio as tio
tensor = torch.rand(1, 16, 16, 1)
transform = tio.CropOrPad((12, 12, 1))
transformed = transform(tensor)
assert transformed.shape == (1, 12, 12, 1)
```

Do you think you can share those images?

**Comment by fepegar** [comment]
Nevermind. I can reproduce with this code:

```python
image = np.random.rand(1, 16, 16, 1)
mask = np.zeros_like(image, dtype=bool)
mask[0, 7, 0] = True
subject = tio.Subject(
    image=tio.ScalarImage(tensor=image),
    mask=tio.LabelMap(tensor=mask),
)
transform = tio.CropOrPad((12, 12, 1), mask_name='mask')
transformed = transform(subject)
assert transformed.shape == (1, 12, 12, 1)
```

I'll investigate, unless @GReguig would like to take a look.

**Comment by fepegar** [comment]
It was a rounding issue.

@all-contributors please add @Linardos for bug.

It will be fixed in `v0.18.23`.
